viewer/src/main/webapp/viewer-html/components/ExternalLinks.js

```diff
@@ -0,0 +1,123 @@
+    menu: null,
+    linkPrefix: "link",
+    config: {
+        urls: null
+    },
+    /**
+     * @constructor
+     */
+    constructor: function (conf) {
+        this.initConfig(conf);
+        viewer.components.ExternalLinks.superclass.constructor.call(this, this.config);
+        var me = this;
+        // Get control of the right-click event:
+        document.oncontextmenu = function (e) {
+            e = e ? e : window.event;
+            me.openMenu(e);
+            if (e.preventDefault)
+                e.preventDefault(); // For non-IE browsers.
+            else {
+                return false; // For IE browsers.
+            }
+        };
+        //TODO don't set SRS hardcoded        
+        Proj4js.defs["EPSG:28992"] = "+proj=sterea +lat_0=52.15616055555555 +lon_0=5.38763888888889 +k=0.9999079 +x_0=155000 +y_0=463000 +ellps=bessel +towgs84=565.237,50.0087,465.658,-0.406857,0.350733,-1.87035,4.0812 +units=m +no_defs";
+        Proj4js.defs["EPSG:4236"] = "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs ";
+
+    },
+    openMenu: function (e) {
+        if (this.menu !== null) {
+            this.menu.destroy();
+        }
+        var x = e.clientX;
+        var y = e.clientY;
+        this.createLinks(x,y);
+        this.menu.showAt(x, y);
+    },
+    createLinks: function (x, y) {
+        var links = [];
+        for (var i = 0; i < this.config.urls.length; i++) {
+            links.push(this.createLink(this.config.urls[i], i, x, y));
+        }
+        var me = this;
+        this.menu = Ext.create("Ext.menu.Menu", {
+            floating: true,
+            items: links,
+            listeners: {
+                click: function (menu, item, e, opts) {
+                    if (!item) {
+                        return;
+                    }
+                    var url = item.config.url;
+                    window.open(url, "_blank");
+                },
+                hide: {
+                    scope: me,
+                    fn: this.hideFired
+                },
+                scope: me
+            }
+        });
+    },
+    createLink: function (link, index, x, y) {
+        var hasCoordsInLink = link.url.indexOf("[") !== -1;
+        var config = {
+            id: this.linkPrefix + index,
+            text: link.label,
+            url: this.procesURL(link.url,x,y),
+            icon: contextPath + (hasCoordsInLink ? "/resources/images/world_link.png" : "/resources/images/link.png")
+        };
+        return config;
+    },
+    procesURL: function (url, x, y) {
+        var map = this.config.viewerController.mapComponent.getMap();
+        var coord = map.pixelToCoordinate(x, y);
+        var extent = map.getExtent();
+        var dest = new Proj4js.Proj("EPSG:4236");
+        var source = new Proj4js.Proj("EPSG:28992");
+        var wgs = new Proj4js.Point(coord.x,coord.y);
+        Proj4js.transform(source,dest,wgs);
+        
+        var currentVariables = {
+            X: coord.x,
```

Comment:
we may want to round these values to realistic values, eg. 1/10th meter and or 1/1000th degrees